src/sidebar/services/oauth-auth.js

```diff
@@ -167,9 +167,27 @@ function auth($http, $rootScope, $window, OAuthClient,
     if (!tokenInfoPromise) {
       var cfg = serviceConfig(settings);
 
+      if (cfg && cfg.grantToken === 'postMessage') {
+        tokenInfoPromise = new Promise(function(resolve, reject) {
+          /** Receive postMessage() response with grant token from LMS app. */
+          function receiveMessage(event) {
+            if (event.origin !== "http://localhost:8001") {
```

Comment:
`"http://localhost:8001"` is the origin of my local dev LMS app. In production this should be `"https://lms.hypothes.is"`. It should be configurable.